care/facility/models/mixins/permissions/patient.py

```diff
@@ -148,7 +149,10 @@ def has_object_update_permission(self, request):
         if (
             request.user.user_type == User.TYPE_VALUE_MAP["DistrictReadOnlyAdmin"]
             or request.user.user_type == User.TYPE_VALUE_MAP["StateReadOnlyAdmin"]
-            or request.user.user_type == User.TYPE_VALUE_MAP["StaffReadOnly"]
         ):
             return False
-        return self.has_object_read_permission(request)
+        return (
+            self.has_object_read_permission(request)
+            and request.user.user_type == User.TYPE_VALUE_MAP["Doctor"]
+            or request.user.user_type >= User.TYPE_VALUE_MAP["DistrictAdmin"]
```

Comment:
This overrides the check for dist admins right? `False and False or True is True`

Reply:
Yup, I messed up the logic 😅